pkg/cmd/adm/register_member_test.go

```diff
@@ -589,104 +412,109 @@ func TestRegisterMember(t *testing.T) {
-				test(t, clusterType, "", true, "--lets-encrypt")
-			})
-			t.Run("multiple toolchains in cluster", func(t *testing.T) {
-				test(t, clusterType, "asdf", false, "--multi-member", "asdf")
-			})
-			t.Run("multiple toolchains in cluster with letsencrypt", func(t *testing.T) {
-				test(t, clusterType, "42", true, "--multi-member", "42", "--lets-encrypt")
-			})
-		})
+func mockCreateToolchainClusterInNamespaceWithReadyCondition(fakeClient *test.FakeClient, namespace string) {
+	fakeClient.MockCreate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.CreateOption) error {
+		if obj, ok := obj.(*toolchainv1alpha1.ToolchainCluster); ok {
+			// force the ready condition on the member toolchaincluster in host cluster only
+			if obj.GetNamespace() == namespace {
+				obj.Status = toolchainv1alpha1.ToolchainClusterStatus{
+					Conditions: []toolchainv1alpha1.Condition{
+						{
+							Type:   toolchainv1alpha1.ConditionReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				}
+			}
+		}
+		return fakeClient.Client.Create(ctx, obj, opts...)
 	}
 }
 
-func TestRunAddClusterScriptFailed(t *testing.T) {
-	// given
-	SetFileConfig(t, Host(), Member())
-	hostKubeconfig := PersistKubeConfigFile(t, HostKubeConfig())
-	memberKubeconfig := PersistKubeConfigFile(t, MemberKubeConfig())
-	gock.New(AddClusterScriptDomain).
-		Get(AddClusterScriptPath).
-		Persist().
-		Reply(404)
-	defer gock.OffAll()
-
-	for _, clusterType := range configuration.ClusterTypes {
-		t.Run("for cluster name: "+clusterType.String(), func(t *testing.T) {
-			// given
-			expArgs := []string{"--type", clusterType.String(), "--host-kubeconfig", hostKubeconfig, "--host-ns", "host-ns", "--member-kubeconfig", memberKubeconfig, "--member-ns", "member-ns", "--lets-encrypt"}
-			ocCommandCreator := NewCommandCreator(t, "echo", "bash",
-				AssertFirstArgPrefixRestEqual("(.*)/add-cluster-(.*)", expArgs...))
-			term := NewFakeTerminalWithResponse("Y")
-
-			// when
-			err := runAddClusterScript(term, ocCommandCreator, clusterType, hostKubeconfig, "host-ns", memberKubeconfig, "member-ns", "", true)
-
-			// then
-			require.Error(t, err)
-			assert.NotContains(t, term.Output(), "Command to be called")
-		})
+func mockCreateToolchainClusterWithReadyCondition(fakeClient *test.FakeClient) {
+	fakeClient.MockCreate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.CreateOption) error {
+		if obj, ok := obj.(*toolchainv1alpha1.ToolchainCluster); ok {
+			obj.Status = toolchainv1alpha1.ToolchainClusterStatus{
+				Conditions: []toolchainv1alpha1.Condition{
+					{
+						Type:   toolchainv1alpha1.ConditionReady,
+						Status: corev1.ConditionTrue,
+					},
+				},
+			}
+		}
+		return fakeClient.Client.Create(ctx, obj, opts...)
 	}
 }
 
+func verifyToolchainClusterSecret(t *testing.T, fakeClient *test.FakeClient, secretNamespace, accessNamespace, tcName string) {
```

Comment:
`accessNamespace` is a bit misleading. Let's name it `ctxNamespace`

Reply:
done in https://github.com/kubesaw/ksctl/pull/53/commits/d31ec759a1c4a9c7d3a654f743930f9ce8b41bc9